### PR TITLE
Add shard-row-id-bits.md doc and fix some links

### PR DESCRIPTION
**Pull request by yikeke**
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/3770/
- Other reference link(s): Fix the dead link from https://github.com/pingcap/docs/pull/3215; might bring conflict to https://github.com/pingcap/docs/pull/3152

### CODE REVIEW

**Comment by nullnotnil** [comment]
This will conflict with https://github.com/pingcap/docs/pull/3152 - can we merge 3152 first, and then this one?

**Comment by yikeke** [comment]
> This will conflict with #3152 - can we merge 3152 first, and then this one?

The one is only relocating the content to a new file and can be merged quickly. I suppose #3152 would have a longer reviewing process, so I prefer merging this PR first? Is that okay? @nullnotnil

**Comment by nullnotnil** [comment]
> This PR is only relocating the content to a new file and can be merged quickly. I suppose #3152 would have a longer reviewing process, so I prefer merging this PR first? Is that okay? @nullnotnil

It is ready to go. It is large in files it touches, but I don't see why review should take long. I have already merged conflicts from another PR that came later (#3215), if I keep having to do that it is going to discourage making improvements to documents.

**Comment by yikeke** [comment]
> > This PR is only relocating the content to a new file and can be merged quickly. I suppose #3152 would have a longer reviewing process, so I prefer merging this PR first? Is that okay? @nullnotnil
> 
> It is ready to go. It is large in files it touches, but I don't see why review should take long. I have already merged conflicts from another PR that came later (#3215), if I keep having to do that it is going to discourage making improvements to documents.

Actually I was gonna add my commits in this PR to https://github.com/pingcap/docs/pull/3215 to fix the deadlink check back then. But something else delayed me for a few hours and when I realized https://github.com/pingcap/docs/pull/3215 was merged😂 with one LGTM.

Please note that according to the new docs collaboration process, we should try to merge every PR after it gets at least two LGTMs - one from the R&D side and one from the language/format side (some PRs should also get an LGTM from DBA's side). We want to take everything a bit slower so we can make things right once. Does this make sense to you? @nullnotnil 

As regard to merging https://github.com/pingcap/docs/pull/3239 or https://github.com/pingcap/docs/pull/3152 first, it's both okay for me (the conflict will be easy to resolve for both PRs). Just a note from my experience, larger changes take longer to get two or three LGTMs.

File: shard-row-id-bits.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+title: SHARD_ROW_ID_BITS
+summary: Learn the SHARD_ROW_ID_BITS attribute.
+---
+
+# SHARD_ROW_ID_BITS
+
+This document introduces the `SHARD_ROW_ID_BITS` table attribute, which is used to set the number of bits of the implicit sharded `_tidb_rowid`.
```
Comment (ran-huang) [inline shard-row-id-bits.md:8]: implicit or implicitly?

File: shard-row-id-bits.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+## Concept
+
+For the tables with a non-integer primary key or no primary key, TiDB uses an implicit auto-increment row ID. When a large number of `INSERT` operations are performed, the data is written into a single Region, causing a write hot spot.
```
[the remark this answers is not in the document]
Comment (yikeke) [reply shard-row-id-bits.md:12]: I updated it to "hot spot".

[the remark this answers is not in the document]
Comment (yikeke) [reply shard-row-id-bits.md:12]: It's not an issue that affects understanding, so I suggest we update this usage case by case.

File: shard-row-id-bits.md (hunk 1 above)
Comment (yikeke) [reply shard-row-id-bits.md:8]: PTAL @kissmydb

Comment (yikeke) [reply shard-row-id-bits.md:8]:
```suggestion
This document introduces the `SHARD_ROW_ID_BITS` table attribute, which is used to set the number of bits of the shards after the implicit `_tidb_rowid` is sharded.
```
Confirmed with @kissmydb , the correct understanding is "表属性 `SHARD_ROW_ID_BITS` 用来设置隐式 `_tidb_rowid` 分片后，分片数量的 bit 位数". PTAL at my suggested edit. @ran-huang @kissmydb

Comment (ran-huang) [reply shard-row-id-bits.md:8]: LGTM

File: best-practices/high-concurrency-best-practices.md [hunk 3]
```diff
@@ -181,7 +181,7 @@ In this case, the table is simple. In other cases, you might also need to consid
 
 **Problem one:**
 
-If a table does not have a primary key, or the primary key is not the `Int` type and you do not want to generate a randomly distributed primary key ID, TiDB provides an implicit `_tidb_rowid` column as the row ID. Generally, when you do not use the `SHARD_ROW_ID_BITS` parameter, the values of the `_tidb_rowid` column are also monotonically increasing, which might causes hotspots too. Refer to [`SHARD_ROW_ID_BITS`](/sql-statements/sql-statement-create-table.md#shard_row_id_bits) for more details.
+If a table does not have a primary key, or the primary key is not the `Int` type and you do not want to generate a randomly distributed primary key ID, TiDB provides an implicit `_tidb_rowid` column as the row ID. Generally, when you do not use the `SHARD_ROW_ID_BITS` parameter, the values of the `_tidb_rowid` column are also monotonically increasing, which might causes hotspots too. Refer to [`SHARD_ROW_ID_BITS` description](/shard-row-id-bits.md) for more details.
```
Comment (yikeke) [inline best-practices/high-concurrency-best-practices.md:184]:
```suggestion
If a table does not have a primary key, or the primary key is not the `Int` type and you do not want to generate a randomly distributed primary key ID, TiDB provides an implicit `_tidb_rowid` column as the row ID. Generally, when you do not use the `SHARD_ROW_ID_BITS` parameter, the values of the `_tidb_rowid` column are also monotonically increasing, which might causes hotspots too. Refer to [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md) for more details.
```

File: faq/tidb-faq.md [hunk 4]
```diff
@@ -1089,7 +1089,7 @@ See [The TiDB Command Options](/command-line-flags-for-tidb-configuration.md).
 
 #### How to scatter the hotspots?
 
-In TiDB, data is divided into Regions for management. Generally, the TiDB hotspot means the Read/Write hotspot in a Region. In TiDB, for the table whose primary key (PK) is not an integer or which has no PK, you can properly break Regions by configuring `SHARD_ROW_ID_BITS` to scatter the Region hotspots. For details, see the introduction of `SHARD_ROW_ID_BITS` in [`CREATE TABLE`](/sql-statements/sql-statement-create-table.md#shard_row_id_bits).
+In TiDB, data is divided into Regions for management. Generally, the TiDB hotspot means the Read/Write hotspot in a Region. In TiDB, for the table whose primary key (PK) is not an integer or which has no PK, you can properly break Regions by configuring `SHARD_ROW_ID_BITS` to scatter the Region hotspots. For details, see the introduction of `SHARD_ROW_ID_BITS` in [TiDB Specific System Variables and Syntax](/shard-row-id-bits.md).
```
Comment (yikeke) [inline faq/tidb-faq.md:1092]:
```suggestion
In TiDB, data is divided into Regions for management. Generally, the TiDB hotspot means the Read/Write hotspot in a Region. In TiDB, for the table whose primary key (PK) is not an integer or which has no PK, you can properly break Regions by configuring `SHARD_ROW_ID_BITS` to scatter the Region hotspots. For details, see the introduction of [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md).
```

**Comment by yikeke** [comment]
I‘ve resolved all the conflicts caused by https://github.com/pingcap/docs/pull/3152. This is ready for review. PTAL @nullnotnil

**Pull request merged**